### PR TITLE
Auto-install gawk on OS-X and Ubuntu if necessary

### PR DESCRIPTION
**Pull request by kigster**
This simplifies README instructions and will help with wider adoption.

In this PR also fixing shebang, otherwise it doesn't work either locally or on CI.

### CODE REVIEW

File: shdoc [hunk 1]
```diff
@@ -62,7 +63,7 @@ function render_toc_link(title) {
     if (style == "github") {
       # https://github.com/jch/html-pipeline/blob/master/lib/html/pipeline/toc_filter.rb#L44-L45
       url = tolower(url)
-      gsub(/[^[:alnum:] -]/, "", url)
+      gsub(/[^[:alnum:] _-]/, "", url)
```
Comment (kigster) [inline shdoc:66]: Huh, I am confused where these changes came from. I did not modify the AWK script in this PR, except the first three lines....

**Comment by kovetskiy** [comment]
Thank you for the contribution, but I don't think the Makefile supposed to install the package's dependencies. Please use brew for this case.

**Comment by kigster** [comment]
That seems weird: 

Makefile IS used in many repos to install dependencies and build the project. 

Makefile has a built in dependency tracking mechanism that avoids doing unnecessary steps. 

As a matter of a fact, it is quite well suited for installing dependencies. Not to mention that brew is OS-X specific. 

It doesn't matter though.

A repo that takes two years to reject contributions on dubious grounds won't be getting my help in the future.

**Comment by kovetskiy** [comment]
No worries, see ya

**Pull request closed without merging**